src/aequitas/bias.py

```diff
@@ -624,14 +663,22 @@ def _get_statistical_significance(cls, original_df, disparity_df, ref_dict,
                 'get_statistical_significance: statistical significance was '
                 'not calculated. There are non-string cols within attr_cols.')
 
+        if not selected_significance:
+            binary_inclusions = {f'binary_{col}' for col in cls.significance_cols}
+        else:
+            binary_inclusions = {f'binary_{col}' for col in selected_significance}
 
         # if no score_thresholds are provided, we assume that rank_abs equals
         # the number  of 1s in the score column; it also serves as flag to set
         # parameter to 'binary'
 
         count_ones = None
         if not score_thresholds:
-            original_df['score'] = original_df['score'].astype(float)
+            with warnings.catch_warnings(record=True) as w:
+                warnings.filterwarnings("ignore",
+                                        message="A value is trying to be set on a copy of a slice from a DataFrame.\nTry using .loc[row_indexer,col_indexer] = value instead")
+                original_df.loc[:, 'score'] = original_df.loc[:, 'score'].astype(float)
```

Comment:
I know we talked about this, and I don't have a solution off the top of my head. But man, this bites.

@saleiro do you know what Pandas's problem is, and what to do about it here?